### PR TITLE
Change VK_EXT_debug_utils to instance extension.

### PR DESCRIPTION
**Pull request by rschiu**
Bug: 149437509

### CODE REVIEW

File: core/vulkan/vk_api_timing_layer/cc/timing_layer.tmpl [hunk 1]
```diff
@@ -219,16 +219,81 @@ VkResult vkDebugMarkerSetObjectNameEXT(
...
 
-static const uint32_t NUM_EXTENSIONS = 2;
-static const VkExtensionProperties new_device_extensions[] = {
+
+namespace {
+
+#define ARRAY_SIZE(a) (sizeof(a) / sizeof(a[0]))
+
+const char* LAYER_NAME = "CPUTiming";
+const uint32_t NUM_EXTENSIONS = 1;
```
Comment (lpy) [inline core/vulkan/vk_api_timing_layer/cc/timing_layer.tmpl]: this one is not used?

File: core/vulkan/vk_api_timing_layer/cc/timing_layer.tmpl [hunk 2]
```diff
@@ -219,16 +219,81 @@ VkResult vkDebugMarkerSetObjectNameEXT(
...
+
+VkResult vkEnumerateDeviceExtensionProperties(
+        PFN_vkEnumerateDeviceExtensionProperties next,
+        VkPhysicalDevice physicalDevice,
+        char const* pLayerName,
+        uint32_t* pPropertyCount,
+        VkExtensionProperties* pProperties) {
     timer t("VkPhysicalDevice", "vkEnumerateDeviceExtensionProperties");
+
+    static const uint32_t NUM_EXTENSIONS = ARRAY_SIZE(NEW_DEVICE_EXTENSIONS);
```
Comment (lpy) [inline core/vulkan/vk_api_timing_layer/cc/timing_layer.tmpl]: maybe we should have `NUM_INSTANCE_EXTENSIONS` and `NUM_DEVICE_EXTENSIONS`, or do `const uint32_t num_extensions = ARRAY_SIZE()` here.

File: core/vulkan/vk_api_timing_layer/cc/timing_layer.tmpl [hunk 3]
```diff
@@ -219,16 +219,81 @@ VkResult vkDebugMarkerSetObjectNameEXT(
...
+ * Enumerate extension properties for a specific layer.
+ *
+ * This should expose only the new extensions added by the layer.
+ */
+VkResult enumerateExtensionPropertiesForLayer(
+        uint32_t* pPropertyCount,
+        VkExtensionProperties* pProperties,
+        uint32_t numExtensions,
+        const VkExtensionProperties extensions[]) {
+    if (pProperties == NULL) {
```
Comment (lpy) [inline core/vulkan/vk_api_timing_layer/cc/timing_layer.tmpl]: nullptr

File: core/vulkan/vk_api_timing_layer/cc/timing_layer.tmpl (hunk 1 above)
Comment (rschiu) [reply core/vulkan/vk_api_timing_layer/cc/timing_layer.tmpl]: Done

File: core/vulkan/vk_api_timing_layer/cc/timing_layer.tmpl (hunk 2 above)
Comment (rschiu) [reply core/vulkan/vk_api_timing_layer/cc/timing_layer.tmpl]: Done

File: core/vulkan/vk_api_timing_layer/cc/timing_layer.tmpl (hunk 3 above)
Comment (rschiu) [reply core/vulkan/vk_api_timing_layer/cc/timing_layer.tmpl]: Done

**Pull request merged**